docs/core/concepts/results.md

```diff
@@ -59,33 +50,54 @@ The most common scenario in which a user might need to directly interact with a
+
+Each of these methods depend on the `Result.location` attribute to determine where the data should actually be persisted. This `location` attribute is the only value serialized to the database of Prefect Core's server or Prefect Cloud. By default this location is autogenerated by a result's `default_location` method, but it can be provided as a formattable string by the user to configure exactly where the result is located.
 
-Result handlers are a more public entity than the `Result` class, as they provide the means to persist the data from an in-memory `Result` object. A result handler is a specific implementation of a `read` / `write` interface for handling data. The only actual requirement for a result handler implementation is that the `write` method returns a JSON-compatible object. 
+### Persisting user-created `Result`s
+If you want to use the `Result` API yourself, such as in a task, instantiate a configured result object and use its `read`, `write`, and `exists` methods directly however you want to in your task.
 
-In general, however, result handler `write` methods are implemented so that they do two things:
-1. serialize the return value of a Task in-memory in the Task State's `Result.value` attribute, and persist it to disk in the result handler's storage backend, and
-2. segregate out a "safe value", usually metadata, about that stored return value in the State's `Result.safe_value` attribute and, for Cloud customers, persist that safe value in the scheduler database.
+For example, if your task needs to read a file from an S3 bucket, consider using Prefect's `S3Result` to identify the S3 file and interact with it:
 
-On the flip side, result handler `read` methods usually access the metadata in an upstream `Result.safe_value` to locate, and then deserialize, the upstream `Task`'s return value that is in storage.
+```python
+from prefect import Flow, task
+from prefect.engine.results import S3Result
+
+@task
+def my_task():
+    s3_result = S3Result(bucket='bucket_o_models')
+    my_saved_model_result = s3_result.read(location='model.pickle')
+    my_model = my_saved_model_result.value
+    # ...
+```
 
-For example, we can easily imagine different kinds of result handlers:
+### Pipeline-persisted `Result`s
 
-- a Google Cloud Storage handler which writes a given piece of data to a Google Cloud Storage bucket, and reads data from that bucket; the `write` method in this instance returns a URI
-- a `LocalResultHandler` that reads / writes data from local file storage; the `write` method in this instance returns an absolute file path
+If checkpointing is turned on, the Prefect pipeline will persist the return value of tasks using the `write` method of the task's configured `Result` subclass. Consider the following example task configured to store its return value as a `LocalResult` into the `~/Desktop/HelloWorld/results` directory. 
```

Comment:
I think there should be a direct link to a checkpoint section in another doc or a tip here on how to enable it

Reply:
@joshmeek +1 on this, maybe we should write an idiom for it and link to the idiom (in a future PR)